File: dto-mapping-benchmark/src/main/java/com/example/dtomappingbenchmark/v2/service/PostService.java
package com.example.dtomappingbenchmark.v2.service;

import com.example.dtomappingbenchmark.v2.entity.Post;
import com.example.dtomappingbenchmark.v2.entity.User;
import com.example.dtomappingbenchmark.v2.repository.PostRepository;
import com.example.dtomappingbenchmark.v2.repository.UserRepository;
import com.example.dtomappingbenchmark.v2.dto.PostDto;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class PostService {
    private final PostRepository postRepository;
    private final UserRepository userRepository;

    public PostDto create(PostDto dto) {
        User user = userRepository.findById(dto.getUserId()).orElseThrow();
        Post post = toEntity(dto, user);
        return toDto(postRepository.save(post));
    }

    PostDto toDto(Post post) {
        return PostDto.builder()
                .id(post.getId())
                .title(post.getTitle())
                .content(post.getContent())
                .userId(post.getUser().getId())
                .build();
    }

    Post toEntity(PostDto dto, User user){
        return Post.builder()
                .title(dto.getTitle())
                .content(dto.getContent())
                .user(user)
                .build();
    }

}

File: object-mapping-benchmark-v5/src/main/kotlin/yu/selab/cj/objectmappingbenchmarkv5/post/service/PostService.kt
package yu.selab.cj.objectmappingbenchmarkv5.post.service

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import yu.selab.cj.objectmappingbenchmarkv5.mapper.PostMapper
import yu.selab.cj.objectmappingbenchmarkv5.post.command.CreatePostDto
import yu.selab.cj.objectmappingbenchmarkv5.post.command.ReadPostDto
import yu.selab.cj.objectmappingbenchmarkv5.post.entity.Post
import yu.selab.cj.objectmappingbenchmarkv5.post.repository.PostRepository
import yu.selab.cj.objectmappingbenchmarkv5.user.repository.UserRepository
import kotlin.collections.map


@Service
class PostService(
    private val postRepository : PostRepository,
    private val userRepository : UserRepository,
    private val postMapper : PostMapper,
) {

    @Transactional(readOnly = true)
    fun getPostList() = postRepository.findAll().map{
        post -> postMapper.toDto(post)
    }

    @Transactional(readOnly = true)
    fun getPost(id : Long) : ReadPostDto {
        val post = postRepository.findById(id).orElseThrow {
            IllegalArgumentException("게시물 없음")
        }
        return postMapper.toDto(post)
    }

    @Transactional
    fun createPost(dto : CreatePostDto) {
        val user = userRepository.findById(dto.userId).orElseThrow {
            IllegalArgumentException("유저 없음")
        }
        val post = Post(dto.title, dto.content, user)
        postRepository.save(post)
    }

    @Transactional
    fun deletePost(id : Long) {
        postRepository.deleteById(id)
    }

}